File: Spotter.Model/Requests/ReservationInsertRequest.cs
namespace Spotter.Model.Requests
{
    public class ReservationInsertRequest
    {
        public int EventId { get; set; }
        public int TicketTypeId { get; set; }
        public int Quantity { get; set; } = 1;
        public string? Note { get; set; }
    }
}

File: UI/spotter_mobile/lib/core/models/reservation_insert_request.dart
class ReservationInsertRequest {
  final int eventId;
  final int ticketTypeId;
  final int quantity;
  final String? note;

  ReservationInsertRequest({
    required this.eventId,
    required this.ticketTypeId,
    this.quantity = 1,
    this.note,
  });

  Map<String, dynamic> toJson() {
    return {
      'eventId': eventId,
      'ticketTypeId': ticketTypeId,
      'quantity': quantity,
      if (note != null) 'note': note,
    };
  }
}
